scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -319,35 +339,15 @@ class FossId(
                     .checkResponse("list scans for project").data
                 checkNotNull(scans)
 
-                val existingScan = scans.sortedByDescending { it.id }.find { scan ->
-                    // The scans in the server contain the url with the credentials so we have to remove it for the
-                    // comparison. If we don't, the scans won't be matched if the password changes!
-                    val urlWithoutCredentials = scan.gitRepoUrl?.replaceCredentialsInUri()
-                    scan.gitBranch == revision && urlWithoutCredentials == url
-                }
-
-                val scanCode = if (existingScan == null) {
-                    log.info { "No scan found for $url and revision $revision. Creating scan ..." }
-
-                    val newUrl = if (addAuthenticationToUrl) queryAuthenticator(url) else url
-
-                    val scanCode = createScan(projectCode, projectName, newUrl, revision)
-                    log.info { "Initiating data download ..." }
-                    service.downloadFromGit(user, apiKey, scanCode)
-                        .checkResponse("download data from Git", false)
-                    scanCode
+                val scanCode = if (deltaScans) {
+                    checkAndCreateDeltaScan(scans, revision, url, projectCode, projectName)
                 } else {
-                    log.info { "Scan ${existingScan.code} found for $url and revision $revision." }
-
-                    requireNotNull(existingScan.code) {
-                        "FossId returned a null scancode for an existing scan"
-                    }
+                    checkAndCreateScan(scans, revision, url, projectCode, projectName)
```

Comment:
For a more pleasant review experience, could also the factoring-out of this function go to a separate (non-functional-change) commit? That allows a reviewer to only skim over commit that do not change / add logic, and to concentrate on the real changes.